plugins/org.eclipse.n4js.semver/src/org/eclipse/n4js/semver/SEMVERHelper.java

```diff
@@ -0,0 +1,100 @@
+package org.eclipse.n4js.semver;
+
+import java.io.StringReader;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.n4js.semver.SEMVER.NPMVersion;
+import org.eclipse.n4js.semver.SEMVER.SimpleVersion;
+import org.eclipse.n4js.semver.SEMVER.VersionNumber;
+import org.eclipse.n4js.semver.SEMVER.VersionRange;
+import org.eclipse.n4js.semver.SEMVER.VersionRangeConstraint;
+import org.eclipse.n4js.semver.SEMVER.VersionRangeSet;
+import org.eclipse.xtext.parser.IParseResult;
+import org.eclipse.xtext.parser.IParser;
+import org.eclipse.xtext.resource.IResourceServiceProvider;
+
+import com.google.inject.Inject;
+
+/** Helper class to parse SEMVER strings and create instances of the SEMVER language */
+public class SEMVERHelper {
+
+	private final IResourceServiceProvider.Registry serviceProviders;
+
+	private IParser semverParser;
+
+	/**
+	 * Creates a new literal converter that obtains an {@link IParser} for regular expressions from the given registry.
+	 */
+	@Inject
+	public SEMVERHelper(IResourceServiceProvider.Registry serviceProviders) {
+		this.serviceProviders = serviceProviders;
+	}
+
+	/** @return parser to parse SEMVER strings */
+	public IParser getSEMVERParser() {
+		if (semverParser == null) {
+			// no need for sync since we can also use a new semverParser if concurrent access happens by accident
+			IResourceServiceProvider serviceProvider = serviceProviders.getResourceServiceProvider(URI
+					.createURI("a." + SEMVERGlobals.FILE_EXTENSION));
+			semverParser = serviceProvider.get(IParser.class);
```

Comment:
we have a utility method for the above: `N4LanguageUtils#getServiceForContext(String fileExtOfContextLanguage, Class<T> serviceType)`